xarray/core/variable.py

```diff
@@ -412,6 +411,10 @@ def data(self):
         if is_duck_array(self._data):
             return self._data
         elif isinstance(self._data, indexing.ExplicitlyIndexed):
+            if pd.api.types.is_extension_array_dtype(self._data) and isinstance(
+                self._data, PandasIndexingAdapter
+            ):
+                return self._data.array
             return self._data.get_duck_array()
```

Comment:
So `get_duck_array` should handle this by possibly converting to a numpy array. Is that not desired? It is ambiguous, to me at least, if we consider ExtensionArrays a "duck array"

Reply:
Right, at the moment `get_duck_array` on the `PandasIndexingAdapter` returns a numpy array, and the docstring states:

```
The Variable's data as an array. The underlying array type
        (e.g. dask, sparse, pint) is preserved.
```

which means we probably want to conserve the pandas array typing.  But right, maybe the implementation of `get_duck_array` should be updated

Reply:
For now I've just added a few ignores.  I think this issue is a bit separate since it doesn't actually affect the runtime behavior

Reply:
I am moving this inside `PandasIndexingAdapter.get_duck_array`